### ISSUE TITLE
Gaphor 2.6 for Windows not working with current directory (.) in path 

### ISSUE THREAD

**Issue by auphofBSF** [gaphor/gaphor#1001]
Gaphor 2.6 Windows Installer failing,  The portable Windows  Gaphor 2.6 does work

### To Reproduce
1. Download https://github.com/gaphor/gaphor/releases/download/2.6.0/gaphor-2.6.0-installer.exe
2. Install
3. From Start Menu select Gaphor
![image](https://user-images.githubusercontent.com/11743864/133708720-e278bba1-24f5-4977-9e4d-17a48659f464.png)

### Expected behavior
Gaphor should execute like the portable version https://github.com/gaphor/gaphor/releases/download/2.6.0/gaphor-2.6.0-portable.exe

### OS
- [ ] Linux (Please put in notes the specific distro)
- [ ] macOS
- [X] Windows

### Version
Current version on Downloads 

### Additional information
Windows 10,  have tried running as administrator but same error shows

**Comment by danyeaw** [comment]
@auphofBSF Thanks for submitting the issue!

I tried reproducing this with 3 different machines: a Win10 VM, a Win10 work computer, and another computer running Win11 preview. I wasn't able to reproduce it on any of them. Any ideas on why your setup might be different?

Could you please open PowerShell and tell me what `$env:Path` prints out?

I was getting a similar error on one of my computers, but I added a workaround to the make-script.py module to strip off extra `;` at the end of the path. The history of this discussion is here:
https://github.com/pyinstaller/pyinstaller/issues/6100

**Comment by auphofBSF** [comment]
Thank you for the clue, `$env:Path` is as follows and shows a path element as `.` , don't know why that is. If I remove it in the powershell `$env:Path=<path with out the identified element>` and then run `.\gaphor-exe`  it works. Would not think that `.` is an illegal path, certainly peculiar and not sure what would have put that there.  I cannot find it in `Control Panel\System and Security\System\Advanced system settings\Environment Variables\` or in any powershell profile.

I have however found in the registry a double semicolon `;;` which should correlate with the `Control Panel ... Advanced.... Env....`
Searched in registry using 
```
# List PowerShell's Paths
Clear-Host
$Reg = "Registry::HKLM\System\CurrentControlSet\Control\Session Manager\Environment"
(Get-ItemProperty -Path "$Reg" -Name PATH).Path
```
which yields the path with a double semicolon `;;` after the `..choclatey\bin` 
see `C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\Program Files\PowerShell\7\;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\bin;;C:\Program Files (x86)\GnuWin32\bin;C:\Program Files\PuTTY\;C:\HashiCorp\Vagrant\bin;C:\Program Files\CMake\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin`


So possibly just another strange windows anomaly, 🤦‍♂️😣. Possibly `make-script.py` should still work ? I have not looked at the code. @danyeaw your thoughts ? 

as mentioned in the opening . the powershell `$env:Path` shows with a path `;.;` after the `chocolatey\bin`  
as follows

```
C:\ProgramData\Anaconda3;C:\ProgramData\Anaconda3\Library\mingw-w64\bin;C:\ProgramData\Anaconda3\Library\usr\bin;C:\ProgramData\Anaconda3\Library\bin;C:\ProgramData\Anaconda3\Scripts;C:\ProgramData\Anaconda3\bin;C:\ProgramData\Anaconda3\condabin;C:\Program Files\PowerShell\7;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0;C:\windows\System32\OpenSSH;C:\Program Files\PowerShell\7;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\bin;.;C:\Program Files (x86)\GnuWin32\bin;C:\Program Files\PuTTY;C:\HashiCorp\Vagrant\bin;C:\Program Files\CMake\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Users\SLLau\.poetry\bin;C:\Users\SLLau\AppData\Local\Microsoft\WindowsApps;C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\SLLau\AppData\Local\atom\bin;c:\users\SLLau\.pyenv\pyenv-win\bin;c:\users\SLLau\.pyenv\pyenv-win\shims;C:\Users\SLLau\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin;c:\Users\SLLau\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\cli-bin;C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code Insiders\bin
```
replacing `;` with `\n` to make more readable

```
C:\ProgramData\Anaconda3
C:\ProgramData\Anaconda3\Library\mingw-w64\bin
C:\ProgramData\Anaconda3\Library\usr\bin
C:\ProgramData\Anaconda3\Library\bin
C:\ProgramData\Anaconda3\Scripts
C:\ProgramData\Anaconda3\bin
C:\ProgramData\Anaconda3\condabin
C:\Program Files\PowerShell\7
C:\windows\system32
C:\windows
C:\windows\System32\Wbem
C:\windows\System32\WindowsPowerShell\v1.0
C:\windows\System32\OpenSSH
C:\Program Files\PowerShell\7
C:\Program Files\Git\cmd
C:\ProgramData\chocolatey\bin
.
C:\Program Files (x86)\GnuWin32\bin
C:\Program Files\PuTTY
C:\HashiCorp\Vagrant\bin
C:\Program Files\CMake\bin
C:\Program Files\Docker\Docker\resources\bin
C:\ProgramData\DockerDesktop\version-bin
C:\Users\SLLau\.poetry\bin
C:\Users\SLLau\AppData\Local\Microsoft\WindowsApps
C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code\bin
C:\Users\SLLau\AppData\Local\atom\bin
c:\users\SLLau\.pyenv\pyenv-win\bin
c:\users\SLLau\.pyenv\pyenv-win\shims
C:\Users\SLLau\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin
c:\Users\SLLau\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\cli-bin
C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code Insiders\bin
```

**Comment by danyeaw** [comment]
@auphofBSF The current `make-script.py` only checks for the extra `;` at the end of the path, but in this case it sounds like yours is in the middle.

Really, PyGObject should be checking that each path is a valid path, but I am OK adding some workarounds to our `make-script.py`. Maybe we can eventually get the checks pushed upstream.

What do you get when you run `python -c "import os; print(os.environ['PATH'])"`?

**Comment by auphofBSF** [comment]
Thanks @danyeaw  output from `python -c "import os; print(os.environ['PATH'])"`

```
C:\ProgramData\Anaconda3;C:\ProgramData\Anaconda3\Library\mingw-w64\bin;C:\ProgramData\Anaconda3\Library\usr\bin;C:\ProgramData\Anaconda3\Library\bin;C:\ProgramData\Anaconda3\Scripts;C:\ProgramData\Anaconda3\bin;C:\ProgramData\Anaconda3\condabin;C:\Program Files\PowerShell\7;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0;C:\windows\System32\OpenSSH;C:\Program Files\PowerShell\7;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\bin;.;C:\Program Files (x86)\GnuWin32\bin;C:\Program Files\PuTTY;C:\HashiCorp\Vagrant\bin;C:\Program Files\CMake\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Users\SLLau\.poetry\bin;C:\Users\SLLau\AppData\Local\Microsoft\WindowsApps;C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\SLLau\AppData\Local\atom\bin;c:\users\SLLau\.pyenv\pyenv-win\bin;c:\users\SLLau\.pyenv\pyenv-win\shims;C:\Users\SLLau\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin;c:\Users\SLLau\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\cli-bin;C:\Users\SLLau\AppData\Local\Programs\Microsoft VS Code Insiders\bin
```

**Comment by danyeaw** [comment]
@auphofBSF Would you have the capability to test if #1022 fixes your issue?

**Comment by danyeaw** [comment]
Closed by #1022.
